kiwi/builder/disk.py

```diff
@@ -229,7 +234,7 @@ def create_disk(self) -> Result:
 
         # a list of instances with the sync_data capability
         # representing the custom partitions area of the disk
-        system_custom_parts: List[FileSystemBase] = []
+        system_custom_parts: Dict[str, FileSystemBase] = {}
```

Comment:
Was this list expected to be in some order? I saw in some parts a `sort` is applied over keys, however keys are strings, hence I assume after sorting keys the results is not in the same order as it was the former `List[FileSystemBase]`.

I am just wondering there could be an error caused by looping through partitions in a different order to what was formerly expected.

Reply:
> Was this list expected to be in some order?

This is a good question. Actually the former list and now the dict contains the FileSystem objects matching each ```<partition>``` element from ```<partitions>```. When syncing the data into the partition device the order does not
matter. However, I agree with you the loop should be sorted such that the iteration is deterministic

While checking on your concern I found there can be another problem caused when using a nested partition
setup. Assuming the following setup:

```xml
<partitions>
    <partition name="usr" size="100" mountpoint="/usr" filesystem="ext3"/>
    <partition name="usr_lib" size="100" mountpoint="/usr/lib" filesystem="ext3"/>
</partitions>
```

It doesn't matter in which order ```/usr``` is synced to ```/dev/some-device-a``` and ```/usr/lib``` to ```/dev/some-device-b``` but the ```usr``` should exclude ```usr/lib``` which is currently not implemented for custom partitions. We have that implemented for volumes (lvm and btrfs) but not for the custom partitions.

This is not related to the clone feature imho but an issue on the custom partitions support when using nested partitions

On behalf of your concern I will make sure the loop runs sorted